pkg/provision/providers/qemu/node.go

```diff
@@ -309,3 +325,25 @@ func (p *provisioner) populateSystemDisk(disks []string, clusterReq provision.Cl
 
 	return nil
 }
+
+func (p *provisioner) createMetalConfigISO(state *vm.State, nodeName, config string) (string, error) {
+	isoPath := state.GetRelativePath(nodeName + "-metal-config.iso")
+
+	tmpDir, err := os.MkdirTemp("", "talos-metal-config-iso")
+	if err != nil {
+		return "", err
+	}
+
+	defer os.RemoveAll(tmpDir) //nolint:errcheck
+
+	if err = os.WriteFile(filepath.Join(tmpDir, "config.yaml"), []byte(config), 0o644); err != nil {
+		return "", err
+	}
+
+	_, err = cmd.Run("mkisofs", "-joliet", "-rock", "-volid", "metal-iso", "-output", isoPath, tmpDir)
```

Comment:
I guess we need a check for the binary similar to how we check if other binaries exist

Reply:
oh... it might fail in the CI because of that.

I feel we should split it up and keep it for power devs only (not for users)